runtime/autoload/provider/clipboard.vim

```diff
@@ -30,13 +30,11 @@ function! s:try_cmd(cmd, ...)
   return out
 endfunction
 
-let s:cache_enabled = 1
 if executable('pbcopy')
   let s:copy['+'] = 'pbcopy'
   let s:paste['+'] = 'pbpaste'
   let s:copy['*'] = s:copy['+']
   let s:paste['*'] = s:paste['+']
-  let s:cache_enabled = 0
```

Comment:
I was a bit unclear... The X tools do no caching, but they enable us to cache clipboard contents, as their process lifetime track ownership, see #2703. OS X clipboard work different so the same model can't be used.